dotnet/src/SemanticKernel/CoreSkills/SemanticFunctionConstants.cs

```diff
@@ -137,22 +70,37 @@ internal static class SemanticFunctionConstants
 
 <goal>If its afternoon please Summarize the input, if the input length > 10 and contains ""book"" then Create an outline for a children's book with 3 chapters about a group of kids in a club and summarize it otherwise translate to japanese and email it to Martin.</goal>
 <plan>
-  <function._GLOBAL_FUNCTIONS_.HourNumber setContextVariable=""$HOUR_NUMBER""/>
+  <function._GLOBAL_FUNCTIONS_.HourNumber setContextVariable=""HOUR_NUMBER""/>
   <if condition=""$HOUR_NUMBER greaterthan 12 OR $HOUR_NUMBER equals 12"">
     <function.Everything.Summarize setContextVariable=""SUMMARIZED_INPUT""/>
-    <function._GLOBAL_FUNCTIONS_.Length setContextVariable=""$SUMMARIZED_INPUT_LENGTH""/>
+    <function._GLOBAL_FUNCTIONS_.Length setContextVariable=""SUMMARIZED_INPUT_LENGTH""/>
     <if condition=""$SUMMARIZED_INPUT_LENGTH greaterthan 10 and $SUMMARIZED_INPUT contains 'book'"">
         <function._GLOBAL_FUNCTIONS_.NovelOutline input=""A group of kids in a club called 'The Thinking Caps' that solve mysteries and puzzles using their creativity and logic."" chapterCount=""3"" />
        <function.Everything.Summarize/>
     </if>
-    <else>
+    <else> (dont use elseif)
         <function.LanguageHelpers.TranslateTo input=""$SUMMARIZED_INPUT"" translate_to_language=""Japanese"" setContextVariable=""TRANSLATED_TEXT"" />
         <function.EmailConnector.LookupContactEmail input=""Martin"" setContextVariable=""CONTACT_RESULT"" />
         <function.EmailConnector.EmailTo input=""$TRANSLATED_TEXT"" recipient=""$CONTACT_RESULT""/>
     </else>
   </if>
 </plan><!-- END -->
 
+Create an XML plan step by step, to satisfy the goal given.
```

Comment:
What's the impact of this change?

Reply:
Moving the Rules to the end I got significantly a better resulting plans from LLM and less occasions where it was using the example functions as available ones.